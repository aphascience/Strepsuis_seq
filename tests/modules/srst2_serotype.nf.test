nextflow_process {

    name "Test Process SRST2_SEROTYPE"
    script "modules/srst2_serotype.nf"
    process "SRST2_SEROTYPE"

    test("Should map reads to serotype reference") {

        when {
            params {
                outdir = "tests/results"
            }
            process {
                """
                input[0] = ["test", file("${projectDir}/tests/data/input_S999_R1_001.fastq.gz"), file("${projectDir}/tests/data/input_S999_R2_001.fastq.gz")]
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

}
