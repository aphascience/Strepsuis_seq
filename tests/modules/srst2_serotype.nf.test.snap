{
    "Should map reads to serotype reference": {
        "content": [
            {
                "0": [
                    [
                        "test",
                        "serotype_test__mlst__Ssuis_serotype__results.txt:md5,1e310455a64cc4c76217a9514d44cb61"
                    ]
                ],
                "1": [
                    [
                        "test",
                        "1"
                    ]
                ],
                "serotype_results": [
                    [
                        "test",
                        "serotype_test__mlst__Ssuis_serotype__results.txt:md5,1e310455a64cc4c76217a9514d44cb61"
                    ]
                ],
                "srst2_sero": [
                    [
                        "test",
                        "1"
                    ]
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.7"
        },
        "timestamp": "2025-09-24T11:22:13.689694445"
    }
}